content/articles/encapsulation-in-c++/index.md

```diff
@@ -0,0 +1,167 @@
+When it comes to data hiding vs. data encapsulation, data hiding conceals class data set components, but encapsulation conceals class data sections as well as private functions.
+You must now be familiar with access specifiers to comprehend data hiding. 
+The class data is typically private, whereas the functions are open. 
+The data is buried to protect it from unintentional modification.
+
+### Roles of Access Specifiers in Encapsulation
+
+Access modifiers, also referred to as **access specifiers**, are used in C++ to hide data. Class members can have varying levels of accessibility based on the type of access specifier used. The three types of Access Modifiers namely public Modifiers, Private Modifiers, and Protect Modifiers.
+
+## Private Access Modifiers
+
+When a data component or member function is set private, it can be accessed and cannot be viewed by anybody outside the class. A fault is produced when accessed by an object that is not a member of the class.
+
+## Public Access Modifiers 
+
+Whereas a public access specifier is applied when creating a class, the base class's public data members change to public members of the derived class, and protected elements become protected in the derived class. Still, the base class's secret members remain hidden.
+
+## Protected Access Modifiers
+
+A Protected Access Specifier is a type of unique access specifier. When a data function is secured, it behaves similarly to private and may be accessed by class members. It may also be accessed through any of the class's sub-classes. 
+
+```
+#include <iostream> 
+using namespace std; 
+class Example { 
+    private: 
+        int num; 
+        
+    public:
+        void set(int input1) 
+        { 
+            num = input1; 
+        } 
+    int get() 
+        { 
+            return num; 
+        } 
+}; 
+int main() 
+{ 
+    Example obj; 
+    obj.set(10);
+    cout << obj.get() << endl; 
+    return 0; 
+} 
+```
+
+In the above example, we have encapsulated the var `num` using private access specifier. We have then used the `public` access specifier to make it available from the other classes, which is the `main` class.
+
+### Real Life Examples of Encapsulation
+1. Automated vending machine: Let's say you visited an automatic vending machine and ordered a drink. The machine will process all your requests, and the drink will be served. An automated vending machine is a class in this case. It comprises data (drinks) and operations (service mechanism) encased in a unified vending machine. This is referred to as encapsulation.
+2. Washing machine: What's the purpose of the power button? It turns the machine on in all cases. But have you ever considered the inner workings? It makes no difference until it works properly. That's what encapsulation is all about. The object is wrapped, and the inside features are concealed. The only difference is that the item may be accessed and used. 
```

Comment:
These two examples needs enhancement and have grammatical issues. Consider rephrasing them.